File: 3903-SmallestStableIndexI/3903-SmallestStableIndexI.cpp
// Last updated: 05/09/2026, 20:50:15
class Solution {
public:
    int firstStableIndex(vector<int>& nums, int k) {
        for (int i = 0; i<nums.size(); i++){
            int mxe = *max_element(nums.begin(),nums.begin()+i+1);
            int mne = *min_element(nums.begin()+i,nums.end());
            if ((mxe-mne)<=k){
                return i;
            }
        }
        return -1;
    }
};

File: 3903-SmallestStableIndexI/3903-SmallestStableIndexI.py
# Last updated: 05/09/2026, 12:51:21
class Solution:
    def firstStableIndex(self, nums: list[int], k: int) -> int:

        n = len(nums)

        for i in range(n):
            if max(nums[0:i+1]) - min(nums[i:n]) <= k:
                return i

        return -1